src/SDWaveFile.cpp

```diff
@@ -173,14 +173,48 @@ int SDWaveFile::begin()
   return 1;
 }
 
+int SDWaveFile::findData(int offset) {
+  uint8_t byteread;
+  int count = 0, fileSize;
+  uint32_t data;
+  uint8_t buffer[] = {0, 0, 0, 0};
+
+  fileSize = _file.size() - offset;
+
+  //check if "data" field is present
+  while (count < fileSize) {
+    if (_file.read((void *)&byteread, 1) != 1) {
+      return -1;
+    }
+    for (int i = 0; i <= 2; i++) {
+      buffer[i] = buffer[i + 1];
+    }
+    buffer[3] = byteread;
+    count++;
+
+    if (count >= 4) {
+      data = 0;
+      data += buffer[0] << 24;
+      data += buffer[1] << 16;
+      data += buffer[2] << 8;
+      data += buffer[3];
+      //if find data field read return the offset size of the subchunk1
+      if (data == 0x64617461) {
+        return count;
+      }
+    }
+  }
+  return -1;
+}
+
 int SDWaveFile::read(void* buffer, size_t size)
 {
   uint32_t position = _file.position();
   int read = _file.read(buffer, size);
 
   if (position == 0) {
     // replace the header with 0's
-    memset(buffer, 0x00, sizeof(struct WaveFileHeader));
+    memset(buffer, 0x00, size);
```

Comment:
Instead of `size` here, I think it should be: `sizeof(struct WaveFileHeader) + _headerOffset`

Reply:
mmmm maybe is better remove completely now because is no longer used all the structure in one time but as a single read to the file with different size each time, what do you think?

Reply:
We need it, otherwise there will be some "noise" when the wave file is played because the head is interpreted as audio.

Reply:
ah ok ok i have understand i just looked better, well this could be done simply with ```sizeof(struct WaveFileHeader)``` about ```_headerOffset```, in my opinion is not required because i cut off the fmt's extras byte and send the header as if it is the PCM one, then this https://github.com/arduino-libraries/ArduinoSound/pull/11/files/485f98353ae43feb80a92947e50e3e099005f770#diff-76c1dfd43efb327c34a33ded368fb3c0R210 is done only in the first call of the read (in findDataHeader), let me know if ok eventually i think a way to integrete ```_headerOffset``` part in the structure